core/tee/socket.c

```diff
@@ -136,11 +136,13 @@ static TEE_Result socket_recv(uint32_t instance_id, uint32_t param_types,
 		return TEE_ERROR_BAD_PARAMETERS;
 	}
 
-	va = thread_rpc_shm_cache_alloc(THREAD_SHM_CACHE_USER_SOCKET,
-					THREAD_SHM_TYPE_APPLICATION,
-					params[1].memref.size, &mobj);
-	if (!va)
-		return TEE_ERROR_OUT_OF_MEMORY;
+	if (params[1].memref.size) {
```

Comment:
Shouldn't `READ_ONCE()` be used systematically when accessing a memref?

Reply:
There's no need for that here, `params` is not in shared memory.

Reply:
OK I totally missed that. Fine then.